servicetalk-concurrent-jdkflow/src/main/java/io/servicetalk/concurrent/jdkflow/JdkFlowAdapters.java

```diff
@@ -0,0 +1,200 @@
+        return new PublisherToFlowPublisher<>(publisher);
+    }
+
+    /**
+     * Converts the passed {@link PublisherSource} to a
+     * <a href="https://docs.oracle.com/javase/9/docs/api/java/util/concurrent/Flow.html">JDK Flow</a>
+     * {@link java.util.concurrent.Flow.Publisher}.
+     *
+     * @param source {@link PublisherSource} to convert to a {@link java.util.concurrent.Flow.Publisher}.
+     * @param <T> Type of items emitted from the {@code source} and the returned
+     * {@link java.util.concurrent.Flow.Publisher}.
+     * @return A {@link java.util.concurrent.Flow.Publisher} representation of the passed {@link PublisherSource}.
+     */
+    public static <T> java.util.concurrent.Flow.Publisher<T> toFlowPublisher(PublisherSource<T> source) {
+        return new PublisherSourceToFlowPublisher<>(source);
+    }
+
+    private static final class PublisherToFlowPublisher<T> implements java.util.concurrent.Flow.Publisher<T> {
+        private final PublisherSource<T> source;
+
+        PublisherToFlowPublisher(final Publisher<T> source) {
+            this.source = toSource(requireNonNull(source));
+        }
+
+        @Override
+        public void subscribe(final Subscriber<? super T> subscriber) {
+            source.subscribe(new FlowSubscriber<>(subscriber));
+        }
+    }
+
+    private static final class PublisherSourceToFlowPublisher<T> implements java.util.concurrent.Flow.Publisher<T> {
+        private final PublisherSource<T> source;
+
+        PublisherSourceToFlowPublisher(final PublisherSource<T> source) {
+            this.source = requireNonNull(source);
+        }
+
+        @Override
+        public void subscribe(final Subscriber<? super T> subscriber) {
+            source.subscribe(new FlowSubscriber<>(subscriber));
+        }
+    }
+
+    private static final class FlowSubscriber<T> implements PublisherSource.Subscriber<T> {
+        private final Subscriber<? super T> subscriber;
+
+        FlowSubscriber(final Subscriber<? super T> subscriber) {
+            this.subscriber = requireNonNull(subscriber);
+        }
+
+        @Override
+        public void onSubscribe(final PublisherSource.Subscription subscription) {
+            subscriber.onSubscribe(new Subscription() {
+                @Override
+                public void request(final long n) {
+                    subscription.request(n);
+                }
+
+                @Override
+                public void cancel() {
+                    subscription.cancel();
+                }
+            });
+        }
+
+        @Override
+        public void onNext(final T next) {
+            subscriber.onNext(next);
+        }
+
+        @Override
+        public void onError(final Throwable t) {
+            subscriber.onError(t);
+        }
+
+        @Override
+        public void onComplete() {
+            subscriber.onComplete();
+        }
+    }
+
+    private static final class FlowPublisherToPublisher<T> extends Publisher<T> {
```

Comment:
nit: rename as `FlowToStPublisher` to match the pattern in other class names.

Reply:
sure